packages/offix-datastore/src/storage/adapters/indexeddb/PredicateExecutor.ts

```diff
@@ -0,0 +1,71 @@
+import { PredicateFunction, PredicateExpression, ModelFieldPredicate } from "../../../predicates";
+
+const getIndexedOperations = (predicate: PredicateFunction): any[] => {
+    if (predicate instanceof PredicateExpression) {
+        return predicate.getPredicates()
+            .map(getIndexedOperations);
+    }
+    const mFPredicate = predicate as ModelFieldPredicate;
+    const fieldInfo = mFPredicate.getFieldInfo();
+
+    if (!fieldInfo.index) { return []; }
+
+    return [{
+        key: mFPredicate.getKey(),
+        operator: mFPredicate.getOperator(),
+        value: mFPredicate.getValue(),
+    }];
+}
+
+const getDataForPredicate = (predicate: ModelFieldPredicate) => {
+    if (predicate.getFieldInfo().index) {
+        return []; // use index to retrieve data
+    }
+    return []; // return all
+}
+
+const handleAndExpression = (predicates: PredicateFunction[]) => {
+    let data: any[] = [];
+    // TODO
+    // To reduce search space here
+    // it seems reasonable to use most constraining index
+    // and filter the data returned by most constraining index
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/IDBKeyRange check options here for reference. SQL lite allows much more including joins etc.